source/IntegrationTests/Behaviours/IncomingRequests/GivenWholesaleServicesRequestWithDelegationTests.cs

```diff
@@ -856,71 +1029,55 @@ private Task GivenWholesaleServicesRequestResponseIsReceived<TType>(Guid process
-        wholesaleServicesRequestMessage.Should().NotBeNull();
+        var (message, processId) = AssertServiceBusMessage(
+            senderSpy,
+            (data) => WholesaleServicesRequest.Parser.ParseFrom(data));
 
         using var assertionScope = new AssertionScope();
 
-        wholesaleServicesRequestMessage.GridAreaCodes.Should().BeEquivalentTo(gridAreas);
-        wholesaleServicesRequestMessage.RequestedForActorNumber.Should().Be(requestedForActorNumber);
-        wholesaleServicesRequestMessage.RequestedForActorRole.Should().Be(requestedForActorRole);
+        message.GridAreaCodes.Should().BeEquivalentTo(gridAreas);
+        message.RequestedForActorNumber.Should().Be(requestedForActorNumber);
+        message.RequestedForActorRole.Should().Be(requestedForActorRole);
 
         if (energySupplierId == null)
-            wholesaleServicesRequestMessage.HasEnergySupplierId.Should().BeFalse();
+            message.HasEnergySupplierId.Should().BeFalse();
         else
-            wholesaleServicesRequestMessage.EnergySupplierId.Should().Be(energySupplierId);
+            message.EnergySupplierId.Should().Be(energySupplierId);
 
         if (chargeOwnerId == null)
-            wholesaleServicesRequestMessage.HasChargeOwnerId.Should().BeFalse();
+            message.HasChargeOwnerId.Should().BeFalse();
         else
-            wholesaleServicesRequestMessage.ChargeOwnerId.Should().Be(chargeOwnerId);
+            message.ChargeOwnerId.Should().Be(chargeOwnerId);
 
         if (resolution == null)
-            wholesaleServicesRequestMessage.HasResolution.Should().BeFalse();
+            message.HasResolution.Should().BeFalse();
         else
-            wholesaleServicesRequestMessage.Resolution.Should().Be(resolution);
+            message.Resolution.Should().Be(resolution);
 
-        wholesaleServicesRequestMessage.BusinessReason.Should().Be(businessReason);
+        message.BusinessReason.Should().Be(businessReason);
 
         if (chargeTypes == null)
         {
-            wholesaleServicesRequestMessage.ChargeTypes.Should().BeEmpty();
+            message.ChargeTypes.Should().BeEmpty();
         }
         else
         {
-            wholesaleServicesRequestMessage.ChargeTypes.Should().BeEquivalentTo(chargeTypes.Select(ct => new Energinet.DataHub.Edi.Requests.ChargeType
+            message.ChargeTypes.Should().BeEquivalentTo(chargeTypes.Select(ct => new Energinet.DataHub.Edi.Requests.ChargeType
             {
                 ChargeType_ = ct.ChargeType,
                 ChargeCode = ct.ChargeCode,
             }));
         }
 
-        wholesaleServicesRequestMessage.PeriodStart.Should().Be(period.Start.ToString());
-        wholesaleServicesRequestMessage.PeriodEnd.Should().Be(period.End.ToString());
+        message.PeriodStart.Should().Be(period.Start.ToString());
+        message.PeriodEnd.Should().Be(period.End.ToString());
```

Comment:
```suggestion
        message.PeriodEnd.Should().Be(period.End?.ToString());
```
Period end should be nullable/optional?